.github/workflows/generate_docs.yml

```diff
@@ -0,0 +1,35 @@
+name: Generate Terraform Plugin Documentation
+
+on:
+  pull_request:
+    branches:
+      - main
+      - stable-*
+
+jobs:
+  generate-docs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: 1.x
+
+      - name: Install dependencies
+        run: go get .
+
+      - name: Install tfplugindocs
+        run: go get github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs@latest
+
+      - name: Build
+        run: go build -v ./...
+
+      - name: Generate documentation
+        run: make generatedocs
+
+      - name: Use GITHUB_TOKEN to commit changes
+        run: |
```

Comment:
I would probably keep an approach where commits to docs are only made in case of prep release PR.  So the documents are automatically updated only in case of release. I this way we limit number of commits to those files and unwanted manual tweaks that may occur. In other cases, we can only render docs and ensure that the generation process works without any issue. If issues occur, just give a log to see the root cause of the issue. But, this is only my thought on this process.

Reply:
Considering our discussions, I am proceeding to merge this PR. Should you feel the need for alterations in this workflow, kindly create an issue, and we can work on the fix.